src/hooks/useChatSocket.tsx

```diff
@@ -39,14 +52,29 @@ export const useChatSocket = () => {
             console.log("Waiting to connect");
         })
 
-        newSocket.on(ServerEvents.MATCHED, ({ partnerName }) => {
+        newSocket.on(ServerEvents.MATCHED, async ({ partnerName, partnerPk }) => {
             setPartner(partnerName);
+
+            const partnerPublicK = await importPublicKey(partnerPk);
+            await storeKey(KeyTransaction.PARTNER_PK, partnerPublicK);
+
             setIsWaiting(false);
             console.log("Matched with a Partner");
             setMessages([]);
         });
 
-        newSocket.on(ServerEvents.RECEIVE_MESSAGE, (data) => {
+        newSocket.on(ServerEvents.RECEIVE_MESSAGE, async (data) => {
+
+            const partnerPk = await getKey(KeyTransaction.PARTNER_PK);
+            const selfPk = await getKey(KeyTransaction.SELF_PK);
+            const sharedSecret = await deriveSharedSecret(selfPk!!, partnerPk!!);
+            const decryptedMsg = await decryptMessage(data.body!!, sharedSecret);
```

Comment:
Here as well

Reply:
Oops, sorry about that. Kotlin habits!! 
I put it like that and there were no issues so I didn't notice that I used the wrong syntax. Could you please remove one of the !s ?
Thanks!

Reply:
It would be better if it is corrected before the pull request is merged from your side.

Reply:
makes sense. i will do it